cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtils.java

```diff
@@ -144,4 +146,46 @@ public static void updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus, Kafka
             }
         }
     }
+
+    /**
+     * Updates the {@code KafkaRebalanceStatus} with a new Warning condition based
+     * on the provided {@link Throwable}. If a Warning condition with the same reason
+     * and message already exists, no update is performed.
+     *
+     * @param status    The {@link KafkaRebalanceStatus} object whose conditions will be updated.
+     * @param exception The {@link Throwable} containing the reason and message for the Warning condition.
+     */
+    public static void addWarningCondition(KafkaRebalanceStatus status, Throwable exception) {
+        List<Condition> conditions = status.getConditions() != null ? new ArrayList<>(status.getConditions()) : new ArrayList<>();
+
+        String causeDescription = exception.getCause() != null
+                ? exception.getCause().toString()
+                : "No cause available";
+        Condition newCondition = StatusUtils.buildWarningCondition(causeDescription, exception.getMessage());
+        Condition oldCondition = conditions.stream()
+                .filter(c -> "Warning".equals(c.getType()))
```

Comment:
Surely we have a enum or constant available (via fabric8 or similar) for the type of a warning condition, rather than hard-coding it here?

Reply:
You are right. We should have an enum for this. I didn't find one, so I just created one. Not sure where best to keep it, so I placed it in the `io.strimzi.operator.cluster.operator.resource` package for now.

Reply:
Yeah, you're right there doesn't seem to be an Enum sadly, the type field is just [a string](https://github.com/fabric8io/kubernetes-client/blob/27b121a862eaae600889fa13bb0584fb7101e44e/extensions/knative/model/src/generated/java/io/fabric8/knative/pkg/apis/Condition.java#L84).

Reply:
As discussed in another thread, I'll open an issue for creating an enum for this and using it across the codebase.

Reply:
Create a separate issue for the task here [1]

[1] https://github.com/strimzi/strimzi-kafka-operator/issues/11413